Options.py

```diff
@@ -466,6 +465,103 @@ def __eq__(self, other: typing.Any):
             raise TypeError(f"Can't compare {self.__class__.__name__} with {other.__class__.__name__}")
 
 
+class BossMeta(AssembleOptions):
+    def __new__(mcs, name, bases, attrs):
+        if name != "PlandoBosses":
+            assert "bosses" in attrs, "Please define valid bosses for Bosses"
+            attrs["bosses"] = frozenset((boss.lower() for boss in attrs["bosses"]))
+            assert "locations" in attrs, "Please define valid locations for Bosses"
+            attrs["locations"] = frozenset((location.lower() for location in attrs["locations"]))
+        return super().__new__(mcs, name, bases, attrs)
+
+
+class PlandoBosses(TextChoice, metaclass=BossMeta):
+    """Generic boss shuffle option that supports plando. Format expected is
+    'location1-boss1;location2-boss2;shuffle_type'. If shuffle_type is not provided in the string, will default to null.
+    Must override can_place_boss, which passes a plando boss and location. Check if the placement is valid for your
+    game here."""
+    bosses: typing.ClassVar[typing.Union[typing.Set[str], typing.FrozenSet[str]]]
+    locations: typing.ClassVar[typing.Union[typing.Set[str], typing.FrozenSet[str]]]
+
+    @classmethod
+    def from_text(cls, text: str):
+        # set all of our text to lower case for name checking
+        text = text.lower()
+        if text == "random":
+            import random
+            return cls(random.choice(list(cls.options.values())))
```

Comment:
these are global random calls, rather from the seeded generation, so this seems like it will produce differing outputs with same seed input

Reply:
isn't the global random seeded to the world seed and being used like that when parsing yaml in all other places already anyway? (the additional import is useless though)

Reply:
the calls were correct but i accidentally copied over the local import from my lttp implementation of this class which probably still works fine but it's unnecessary at the least.